consoleme/lib/aws.py

```diff
@@ -1506,3 +1508,114 @@ async def get_scps_for_account_or_ou(identifier: str) -> ServiceControlPolicyArr
+    :param inline_iam_policy_statements: A list of IAM policy statement dictionaries
+    :return: A potentially more compact list of IAM policy statement dictionaries
+    """
+    exclude_ids = []
+    minimized_statements = []
+
+    inline_iam_policy_statements = await normalize_policies(
+        inline_iam_policy_statements
+    )
+
+    for i in range(len(inline_iam_policy_statements)):
+        inline_iam_policy_statement = inline_iam_policy_statements[i]
+        if disregard_sid:
+            inline_iam_policy_statement.pop("Sid", None)
+        if i in exclude_ids:
+            # We've already combined this policy with another. Ignore it.
+            continue
+        for j in range(i + 1, len(inline_iam_policy_statements)):
+            if j in exclude_ids:
+                # We've already combined this policy with another. Ignore it.
+                continue
+            inline_iam_policy_statement_to_compare = inline_iam_policy_statements[j]
+            if disregard_sid:
+                inline_iam_policy_statement_to_compare.pop("Sid", None)
+            # Check to see if policy statements are identical except for a given element. Merge the policies
+            # if possible.
+            for element in [
+                "Resource",
+                "Action",
+                "NotAction",
+                "NotResource",
+                "NotPrincipal",
+            ]:
+                if not (
+                    inline_iam_policy_statement.get(element)
+                    or inline_iam_policy_statement_to_compare.get(element)
+                ):
+                    # This function won't handle `Condition`.
+                    continue
+                diff = DeepDiff(
+                    inline_iam_policy_statement,
+                    inline_iam_policy_statement_to_compare,
+                    ignore_order=True,
+                    exclude_paths=[f"root['{element}']"],
+                )
+                if not diff:
+                    exclude_ids.append(j)
+                    # Policy can be minimized
+                    inline_iam_policy_statement[element] = sorted(
+                        list(
+                            set(
+                                inline_iam_policy_statement[element]
+                                + inline_iam_policy_statement_to_compare[element]
+                            )
+                        )
+                    )
+                    break
+
+    for i in range(len(inline_iam_policy_statements)):
+        if i not in exclude_ids:
+            inline_iam_policy_statements[i] = sort_dict(inline_iam_policy_statements[i])
+            minimized_statements.append(inline_iam_policy_statements[i])
+    # TODO(cccastrapel): Intelligently combine actions and/or resources if they include wildcards
+    minimized_statements = await normalize_policies(minimized_statements)
+    return minimized_statements
+
+
+async def normalize_policies(policies: List[Any]) -> List[Any]:
```

Comment:
Tests for this as well

Reply:
Ditto, mostly covered indirectly but fairly straightforward to write a direct test

![image](https://user-images.githubusercontent.com/7538233/124003037-4ff3ee80-d98b-11eb-8ffb-a21da279a7d8.png)